androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/fragments/ConfirmOrderFragment.java

```diff
@@ -2,34 +2,46 @@
 
 import android.app.AlertDialog;
 import android.app.Dialog;
-import android.arch.lifecycle.ViewModelProviders;
-import android.content.DialogInterface;
 import android.os.Bundle;
-import android.support.v4.app.DialogFragment;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.widget.TextView;
 
 import com.hopsquad.hopsquadapp.R;
-import com.hopsquad.hopsquadapp.activities.MainActivity;
-import com.hopsquad.hopsquadapp.models.Order;
-import com.hopsquad.hopsquadapp.viewmodels.TapListViewModel;
 
 import java.text.NumberFormat;
 
-public class ConfirmOrderFragment extends DialogFragment {
+public class ConfirmOrderFragment extends BaseDialogFragment {
+
+    private static final String TOTAL_PRICE_TO_PAY = "TOTAL_PRICE";
+    private OnConfirmDialogOptionSelectedListener mListener;
+    private float mTotal;
+
+    public interface OnConfirmDialogOptionSelectedListener {
+
+        void onConfirmDialogOptionSelected(boolean confirmed);
+    }
+
+    public static ConfirmOrderFragment newInstance(float total) {
+        ConfirmOrderFragment fragment = new ConfirmOrderFragment();
+        Bundle args = new Bundle();
+        args.putFloat(TOTAL_PRICE_TO_PAY, total);
+        fragment.setArguments(args);
+        return fragment;
+    }
 
     public ConfirmOrderFragment() {
         // Required empty public constructor
     }
 
-    TapListViewModel viewModel;
+    public void setOnConfirmDialogOptionSelectedListener(OnConfirmDialogOptionSelectedListener listener) {
+        this.mListener = listener;
+    }
 
     @Override
     public void onCreate(Bundle savedInstaceState) {
         super.onCreate(savedInstaceState);
-
-        viewModel = ViewModelProviders.of(getTapListFragment()).get(TapListViewModel.class);
+        mTotal = getArguments().getFloat(TOTAL_PRICE_TO_PAY);
```

Comment:
Nice. I did the same for the beer description args. Good to know I am following the right path.